source/drivers/MicroBitCompass.cpp

```diff
@@ -357,6 +358,7 @@ int MicroBitCompass::getPeriod()
  */
 int MicroBitCompass::requestUpdate()
 {
+    microbit_panic(MICROBIT_HARDWARE_UNAVAILABLE_MAG);
     return MICROBIT_NOT_SUPPORTED;
```

Comment:
@microbit-sam and @finneyj do you think we also need to panic in any other functions: is it misleading to allow things like clearCalibration() and configure() to succeed on the instance created when specific hardware isn't available? Same applies to the accelerometer too, I think. 

For real-world cases I think in all of these we're going to call requestUpdate() soon enough and get the actual failure, but are there cases where you can make 'illegitimate progress' in the absence device-specific instance and we should panic sooner?

Reply:
Thanks - yes please Joe!